enhancements/ingress/aws-lb-eip-selection.md

```diff
@@ -0,0 +1,348 @@
+    1. The admin sets the `eipAllocations` in the installer using the install-config.yaml.
+    2. The installer then creates a default Ingress Controller resource with the `eipAllocations`.
+    3. The cluster ingress operator watches the resource and then creates a `default` `IngressController` by:
+        1. creating an `default` `IngressController` from the `default` `Ingress Controller CR` created by the installer.
+        2. then scaling a deployment for the `IngressController` CR which has the value of the `eipAllocations`.
+        3. then creating a service of service type load balancer with the annotation `service.beta.kubernetes.io/aws-load-balancer-eip-allocations`,
+           which uses the value from the field `eipAllocations` of `IngressController` CR.
+
+#### Field Validation
+
+##### Validation on installer when installing in managed VPC (full-automated) based in the discovered zones used to create the cluster.
+   Add a function to the installer code to  compare the number of Availability Zones in the region to the number of eipAllocations passed in the `install-config.yaml`.   This is required because the cluster will select one subnet per AZ, so the number of EIPs must be equal to the number of subnets.
+
+##### The number of public subnets must match the number of `eipAllocations` in an unmanaged VPC (BYO VPC)
+     We will be comparing if the number of public subnets added to the install-config matches with len(eipAllocations).
+
+##### Ensure that EIPs exist and are not already assigned.
+EIPs can be assigned to many resource types, like NAT gateways, load balancers, instances, etc. The attribute `associationId` will be set when the EIP is already associated.
+To mitigate this we could add that validation, at least on installer, to provide quick-feedback (fail when validate install-config) to the user when the provided EIP is already associated to another resource.
+It would be nice to have a validation before setting the annotation to CCM, keeping the operator Degraded before disrupting the service.
+
+##### EIP Allocation Defaulting Mechanics for Ingress Controller
+Traditionally, the Ingress Operator has populated default values from the Ingress Config into the `status`, 
+making `status` effectively reflect the desired state of the IngressController. However, since 
+`eipAllocations` in `status` represents the **actual** state, not the **desired** state, the default 
+`eipAllocations` values must be set in the `spec` when the Ingress Operator initially admits the 
+IngressController.
+This approach is new. The Ingress Operator does not typically set default values in `spec` for load balancer
+configurations if the user hasn’t explicitly provided them. While this defaulting pattern is more consistent
+with Kubernetes conventions for `spec` and `status` (and is also our only option in this situation), it's
+important to acknowledge that this inconsistency in defaulting behavior could cause confusion for users.
```

Comment:
You should remove this section now, we simplified to remove the Ingress Config like you mentioned, so no relevance to EIPAllocations.